ledger/participant-integration-api/src/main/scala/platform/store/backend/common/QueryStrategy.scala

```diff
@@ -59,11 +60,52 @@ trait QueryStrategy {
   /** Boolean predicate */
   def isTrue(booleanColumnName: String): String
 
+  /** Constant boolean to be used in a SELECT clause */
+  def constBoolean(value: Boolean): String
+
   /** ANY SQL clause generation for a number of Long values
     */
   def anyOf(longs: Iterable[Long]): CompositeSql = {
     val longArray: Array[java.lang.Long] =
       longs.view.map(Long.box).toArray
     cSQL"= ANY($longArray)"
   }
+
+  /** Expression for `(offset <= endInclusive)`
+    *
+    *  The offset column must only contain valid offsets (no NULL, no Offset.beforeBegin)
+    */
+  def offsetIsSmallerOrEqual(nonNullableColumn: String, endInclusive: Offset): CompositeSql = {
+    import com.daml.platform.store.Conversions.OffsetToStatement
+    // Note: there are two reasons for special casing Offset.beforeBegin:
+    // 1. simpler query
+    // 2. on Oracle, Offset.beforeBegin is equivalent to NULL and cannot be compared with
+    if (endInclusive == Offset.beforeBegin) {
+      cSQL"#${constBoolean(false)}"
```

Comment:
can it be that this is dead code?
* checked couple of usages, seemed to boil down to Dispatcher calls, those will populate here non-beforeBegin
* previous implementation was having no such treatment

Similarly for line 104

(I wish we would not have an Offset.beforeBegin at all, this is a null concept which does not play nicely with scala :( )

Reply:
I have tried replacing this line with an exception and CI fails. Apparently we do call this method with `endInclusive == Offset.beforeBegin`. The previous implementation happened to also be correct both on Oracle (because `x <= NULL` is always false) and Postgres, but I think the extra treatment is safer.